tests/unit/SearchIndexHelperTest.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace unit\helpers;
+
+use Codeception\Test\Unit;
+use Craft;
+use craft\db\Table;
+use viget\base\helpers\SearchIndex;
+
+class SearchIndexHelperTest extends Unit
+{
+    protected function setUp(): void
+    {
+        parent::setUp();
+        Craft::$app->getDb()->createCommand()->truncateTable(Table::SEARCHINDEX)->execute();
+    }
+
+    public function testOptimizeForMySql()
+    {
+        Craft::$app->getDb()->driverName = 'mysql';
+        $result = SearchIndex::optimize();
+
+        // Assert: Verify that the result is an integer (number of rows affected)
+        $this->assertIsInt($result);
+    }
+
+    public function testOptimizeForPostgres()
+    {
+        Craft::$app->getDb()->driverName = 'pgsql';
+        $result = SearchIndex::optimize();
+
+        // Assert: Verify that the result is an integer (number of rows affected)
+        $this->assertIsInt($result);
```

Comment:
I feel like this might throw an SQL error unless our test environment is using PostgreSQL. 

I'm planning to switch our CI tests to use GitHub actions. So I might be able to make a matrix of tests (one for MySQL one for PostgreSQL).

Reply:
I just merged an updated GitHub actions based test runner. 

It uses a matrix to run tests with MySQL and Postgres. 

It also switches the local dev environment to DDEV.

Reply:
I rebased this work and added test skipping based on db driver so the unit test works with the new matrix strategy.